app/models/course/assessment/question/programming.rb

```diff
@@ -111,11 +117,15 @@ def process_new_package
 
   # Removes the template files and test cases from the old package.
   def remove_old_package
-    template_files.clear
+    template_files.clear unless template_files_changed?
```

Comment:
It was intended to not clear the template files that were assigned to the question when it is not autograded, but it wasn't working correctly. I have changed this to use a boolean flag to indicate non-autograded question instead.